swaps/swaps_service.go

```diff
@@ -207,13 +210,35 @@ func (svc *swapsService) EnableAutoSwapOut() error {
 					logger.Logger.Info("Threshold requirements not met for swap, ignoring")
 					continue
 				}
+
+				actualDestination := swapDestination
+				var addressIndex uint32
+				if swapDestination != "" {
+					if err := svc.validateXpub(swapDestination); err == nil {
+						actualDestination, addressIndex, err = svc.getNextUnusedAddressFromXpub()
+						if err != nil {
+							logger.Logger.WithError(err).Error("Failed to get next address from xpub")
+							actualDestination = "" // Fallback to empty address if we can't get one
```

Comment:
I think we should fail the swap instead. And we should verify the xpub when the user submits the auto swap details